### PR TITLE
feat(cli): properly handle boolean values

### PR DESCRIPTION
**Pull request by bagage**
Part of #3178 

Throw exception if boolean value is invalid.

### CODE REVIEW

File: lib/config/index.js [hunk 1]
```diff
@@ -59,6 +57,8 @@ async function parseConfigs(env, argv) {
   // Set log level
   logger.levels('stdout', config.logLevel);
 
+  logger.debug('Parsing configs');
```
Comment (rarkins) [inline lib/config/index.js]: No, let's revert this. If the user has set `LOG_LEVEL=debug` in env then the message will be logged where it was, and it's good to log "Parsing configs" before the parsing is done.

**Comment by rarkins** [comment]
This is a good idea for reducing chances of unpredictable CLI outcomes

File: lib/config/index.js (hunk 1 above)
Comment (bagage) [reply lib/config/index.js]: Oh, yes indeed. Though it is misleading when you use CLI `--log-level debug` and this log message does not appear.

Comment (rarkins) [reply lib/config/index.js]: It's a little chicken and egg but basically if you need logs of the parsing itself then you need to do it via env, as we have a special rule for that.

**Comment by bagage** [comment]
I cannot fix the broken CI though, I believe there are no test around cli parsing?

**Comment by rarkins** [comment]
Tests are here: https://github.com/renovatebot/renovate/blob/master/test/config/cli.spec.js

**Comment by bagage** [comment]
Indeed @rarkins, my editor is not properly configured for Renovate and I missed them. Added a test for bad value :).

**Pull request merged**